yt_dlp/extractor/lecturio.py

```diff
@@ -16,16 +16,27 @@
 
 class LecturioBaseIE(InfoExtractor):
     _API_BASE_URL = 'https://app.lecturio.com/api/en/latest/html5/'
+    _DE_API_BASE_URL = 'https://lecturio.de/api/de/latest/html5/'
     _LOGIN_URL = 'https://app.lecturio.com/en/login'
+    _DE_LOGIN_URL = 'https://www.lecturio.de/anmelden.html'
     _NETRC_MACHINE = 'lecturio'
 
+    is_DE = None
+
+    # Find out if url is german before starting anything else
+    def extract(self, url):
+        self.is_DE = True if re.match(r"https://(?:www\.)?lecturio\.de/", url) else False
+        return super().extract(url)
```

Comment:
Don't use global state like thi. Pass it as argument through the relevant functions.

Reply:
Is  `_initialize_pre_login` the right place for this? I still don't see how I could pass it as an argument, since  `_perform_login` is only passed the username and password by  `initialize`